src/server/browserType.ts

```diff
@@ -186,7 +186,8 @@ export abstract class BrowserType extends SdkObject {
       await validateHostRequirements(this._registry, options.channel as registry.BrowserName);
 
     let wsEndpointCallback: ((wsEndpoint: string) => void) | undefined;
-    const wsEndpoint = options.useWebSocket ? new Promise<string>(f => wsEndpointCallback = f) : undefined;
+    const shouldWaitForWSListening = options.useWebSocket || options.args?.some(a => a.startsWith('--remote-debugging-port='));
```

Comment:
nit: It is possible to pass two arguments: `--remote-debugging-port`, `1234`. So, let's drop the `=` sign from `startsWith`.